### ISSUE TITLE
False negatives nested at-if in selector-max-id

### ISSUE THREAD

**Issue by BenjaminWFox** [stylelint/stylelint#3105]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where "selector-max-id" is not applied if the selector is followed by an SCSS @if statement.

> Which rule, if any, is this issue related to?

"selector-max-id"

> What CSS is needed to reproduce this issue? (SCSS)

e.g.

```css
$p: 4;

Not a violation:
#temp #two {
  @if ($p == 4) {
    font-size: 4em;
  }
  
  color: #000;
}

A violation:
#temp #two {  
  color: #000;
}

```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "selector-max-id": 1
  }
}
```

> Which version of stylelint are you using?

8.4.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Sublimetext 3 with SublimeLinter-contrib-stylelint
AND
Webpack with stylelint-webpack-plugin

(issue shows up with both)

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS @if

> What did you expect to happen?

selector-max-id should be applied regardless of what follows in the block

> What actually happened (e.g. what warnings or errors you are getting)?

selector-max-id was not applied when the block started with @if

**Comment by jeddy3** [comment]
@BenjaminWFox Thanks for the report and for using the template.

Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) as the core team is focused on standard syntax. I suspect the issue is to [do with resolving nested blocks](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-max-id/index.js#L66-L73). At the moment, only the _fullest_ selector is checked. Perhaps rule-sets should be ignored _only if_ any of its children _contain_ id selectors.

**Comment by BenjaminWFox** [comment]
@jeddy3 Thanks for the pointers. I'll take a look and see what I can come up with.

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/3113
